src/main/scala/com/codacy/plugins/api/DockerApi.scala

```diff
@@ -30,3 +30,22 @@ object Options {
   trait Value extends Any
 
 }
+
+sealed trait ToolResults[T]
+object ToolResults {
+  case class Warning(message: String) {
+    override def toString: String = message
+  }
+  case class ToolSuccessResults[T](results: List[T], warnings: Option[List[Warning]] = None) extends ToolResults[T]
+  case class ToolFailure[T](reason: ToolFailureReason, message: String, stackTrace: Option[String] = None) extends ToolResults[T]
+
+  type ToolFailureReason = ToolFailureReason.Value
+  object ToolFailureReason extends Enumeration {
+    val MissingConfigs: Value = Value("Missing Configurations")
+    val MissingArtifacts: Value = Value("Missing Compilation Artifacts")
+    val UnsupportedConfigs: Value = Value("Unsupported Configurations")
+    val FailedAnalysis: Value = Value("Failed Analysis")
+    val TimedOut: Value = Value("Timed Out")
+    val OtherReason: Value = Value("Other Reason")
```

Comment:
If we're serializing this I'd rather have codes as the values instead of Strings so it's obvious it's a breaking change to touch the values.